Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -488,6 +488,29 @@ class HTTPClientTests: XCTestCase {
         XCTAssertEqual(10, count)
     }
 
+    func testFileDownload() throws {
+        var request = try Request(url: self.defaultHTTPBinURLPrefix + "events/10/1")
+        request.headers.add(name: "Accept", value: "text/event-stream")
+
+        let (totalBytes, receivedBytes) =
+            try TemporaryFileHelpers.withTemporaryUnixDomainSocketPathName { path -> (Int?, Int) in
```

Comment:
This is a bit of a nit, but I'd like us to rename this function. I'm fine with having `withTemporaryUnixDomainSocketPathName` call into an underlying function, but I think it's confusing to claim we're going to use a file as a unix socket and then to use it for something else altogether.